operator/src/main/java/org/bf2/operator/managers/IngressControllerManager.java

```diff
@@ -668,12 +695,31 @@ private int numReplicasForConnectionDemand(double connectionDemand) {
         return (int)Math.ceil(connectionDemand * (peakConnectionPercentage / 100D) / maxIngressConnections);
     }
 
-    private String getIngressControllerDomain(String ingressControllerName) {
-        return ingressControllerInformer.getList().stream()
-                .filter(ic -> ic.getMetadata().getName().equals(ingressControllerName))
-                .map(ic -> ROUTER_SUBDOMAIN + (ic.getStatus() != null ? ic.getStatus().getDomain() : ic.getSpec().getDomain()))
-                .findFirst()
-                .orElse("");
+    private String getIngressControllerDomain(String ingressControllerName, boolean hasPublicDns) {
+      if (!isPrivateNetwork(informerManager.getLocalAgent()) && !hasPublicDns) {
```

Comment:
Optional efficiency nit - 
```suggestion
      if (!hasPublicDns && !isPrivateNetwork(informerManager.getLocalAgent())) {
```